homeassistant/components/thermostat/homematic.py

```diff
@@ -97,9 +132,10 @@ def target_temperature(self):
 
     def set_temperature(self, temperature):
         """Set new target temperature."""
-        self.device.setValue(self._full_device_name,
-                             PROPERTY_SET_TEMPERATURE,
-                             temperature)
+        device = ServerProxy(self.address)
```

Comment:
Tried that, didn't work out that well.
For one, the scope of the lock needs to be as big as the scope of the ServerProxy, that could be done, I guess.
But we would need to be sure that every transaction has been completed and all responses read before starting the next one, and that didn't seem to completely work.
This way it seems to be quite dependable.